src/collection.cpp

```diff
@@ -1557,10 +1557,10 @@ Option<nlohmann::json> Collection::search(std::string  raw_query,
 
     // for grouping we have to aggregate group set sizes to a count value
     if(group_limit) {
-        total_found = search_params->groups_processed.size() + override_result_kvs.size();
-    } else {
-        total_found = search_params->all_result_ids_len;
-    }
+        total_groups_found = search_params->groups_processed.size() + override_result_kvs.size();
+    } 
+    
+    total_found = search_params->all_result_ids_len;
```

Comment:
We can't break existing behavior. `total_found` should remain as the current logic, i.e. number of groups for group queries and number of docs otherwise. Instead lets add `found_docs` as an additional field available for group queries.